torch/_subclasses/functional_tensor.py

```diff
@@ -42,11 +42,45 @@ def __new__(cls, elem):
         kwargs["layout"] = elem.layout
         kwargs["requires_grad"] = elem.requires_grad
         kwargs["dtype"] = elem.dtype
+        kwargs["dispatch_sizes_strides_policy"] = "sizes"
         out = torch.Tensor._make_wrapper_subclass(cls, **kwargs)
         out.elem = elem
         return out
 
+    # Need to disable default torch_function. Why?
+    # Default torch_function will always wrap outputs into a subclass if they aren't already a subclass.
+    # We actually.. don't want to do this sometimes, see Note [FunctionalTensorMode inputs are sometimes plain tensors]
+    __torch_function__ = torch._C._disabled_torch_function_impl
+
     def __torch_dispatch__(self, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        # FunctionalTensor needs to plumb all metadata requests to the inner tensor.
+        # In theory we don't have to do this - but if we want to service metadata requests here,
+        # we need to carefully make sure all metadata is accurate (including metadata mutations)
+        if func in [
+            torch.ops.aten.is_contiguous.default,
+            torch.ops.aten.is_contiguous.memory_format,
+            torch.ops.aten.is_strides_like_format.default,
+            torch.ops.aten.is_non_overlapping_and_dense.default,
+            torch.ops.aten.size.default,
+            torch.ops.aten.sym_size.default,
+            torch.ops.aten.stride.default,
+            torch.ops.aten.sym_stride.default,
+            torch.ops.aten.storage_offset.default,
+            torch.ops.aten.sym_storage_offset.default,
+            torch.ops.aten.numel.default,
+            torch.ops.aten.sym_numel.default,
+            torch.ops.aten.dim.default,
+        ]:
+            def unwrap(x):
+                return x.elem
+            args_unwrapped = pytree.tree_map_only(FunctionalTensor, unwrap, args)
+            # All metadata accesses should be plumbed to the inner tensor, that way we don't have to worry
+            # about the problem of keeping metadata in sync between the wrapper and inner tensor.
+            # This also alleviates us from having to manually handle metadata mutations on the wrapper.
```

Comment:
Or another question is: FunctionalTensorWrapper (in C++) doesn't need to do this. Why are we doing it here?

Reply:
Sorry this q slipped by.

> gah, is there a way for us to know to update this if we add a new metadata field?

FunctionalTensor should error loudly at least - since `FunctionalTensor.__torch_dispatch__()` always raises an error right now (except for this hardcoded list of metadata requests)

> Or another question is: FunctionalTensorWrapper (in C++) doesn't need to do this. Why are we doing it here?

Actually, FunctionalTensorWrapper does this today ([code](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/FunctionalTensorWrapper.cpp#L370)). It overrides all of the overrideable metadata methods on TensorImpl to just plumb them to the inner tensor.